jasmin_cloud/provider/openstack/provider.py

```diff
@@ -646,11 +646,14 @@ def delete_machine(self, machine):
         """
         See :py:meth:`.base.ScopedSession.delete_machine`.
         """
+        caas_machine = True if machine.provisioned_by_caas and machine.provisioned_by_caas == 1 else False
```

Comment:
You don't need this - what you need to do is make sure `machine.provisioned_by_caas` is populated and a boolean, similar to `nat_allowed`.

`machine` is also allowed to be given as just a machine id, so you will need to make sure to convert it to a machine instance in that case. See `attach_external_ip` for an example.